bin/yarn2nix-fix-sha1.js

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env node
+"use strict";
+
+/**
+ * Fixes the yarn.lock to include the sha1 on all entries.
+ *
+ * For some URLs like codeload.github.com, yarn decides not to generate the sha1.
+ */
+
+const fs = require("fs");
+const lockfile = require('@yarnpkg/lockfile');
+const path = require("path");
+const https = require("https");
+const crypto = require('crypto');
+const util = require("util");
+
+function getSha1(url) {
+  return new Promise((resolve, reject) => {
+    https.get(url, (res) => {
+      const { statusCode } = res;
+      const hash = crypto.createHash('sha1');
+      if (statusCode !== 200) {
+        const err = new Error('Request Failed.\n' +
+                          `Status Code: ${statusCode}`);
+        // consume response data to free up memory
+        res.resume();
+        reject(err);
+      }
+
+      res.on('data', (chunk) => { hash.update(chunk); });
+      res.on('end', () => { resolve(hash.digest('hex')) });
+      res.on('error', reject);
+    });
+  });
+};
+
+function updateResolvedSha1(pkg) {
+  let [url, sha1] = pkg.resolved.split("#", 2)
```

Comment:
This fails on local file dependencies. For example, here's a dependency from `yarn.lock` without a `resolved` field:

```
gojs@./lib/gojs:
  version "1.7.14"
```

Returning early works: `if (!pkg.resolved) { return Promise.resolve(); }`

Reply:
thanks! I don't think that local dependencies are supported right now but it's not an excuse to break the generation. Fixed in c2962f26df43511202a9895c2083a19f1aa2fa8c